exporters/dsexporter/dsexporter_test.go

```diff
@@ -1,46 +1,83 @@
 package main
 
 import (
+	"fmt"
 	"testing"
+	"net/http"
+	"net/http/httptest"
 
-	"github.com/prometheus/client_golang/prometheus"
 	"github.com/stretchr/testify/assert"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-func TestCustomCollector(t *testing.T) {
-	exporter := NewCustomCollector()
-	prometheus.MustRegister(exporter)
-
-	// Simulate collecting metrics and check the exported metric value.
-	metrics := prometheus.DefaultGatherer
-	metricFamilies, err := metrics.Gather()
-	assert.NoError(t, err)
-
-	var RequestCountValue float64
-	for _, mf := range metricFamilies {
-		if mf.GetName() == "request_count" {
-			RequestCountValue = mf.GetMetric()[0].GetCounter().GetValue()
-			break
-		}
+func TestAvailabilityHandler(t *testing.T) {
+	mockExporter := NewCustomCollector()
+
+	req, err := http.NewRequest("GET", "/?service=test.com&check=test", nil)
```

Comment:
According to the metric [spec](https://docs.google.com/document/d/193FO8DYP8zCzzued53DiIT84xmyTm51v_OqiWzQi8Bg), the `service` label should be an identifier for the service under evaluation, not the lower level check of its dependency (`test.com` as provided here). It should be a constant value shared across the entire exporter (e.g. `dummy-service`) and not provided dynamically.

Also, wdyt about adding more checks (like `github` which could query `https://api.github.com/`) to make this example more realistic? `test` isn't an obvious enough example on its own IMO.

Reply:
@hmariset I think that having a clear idea (and having documented on the PR description and on the ticket) regarding what sort of metrics we want to include here will be super important for being able to deliver on it.

My take is that when we named it dsexporter, we were planning to use it to export the availability of the grafana data source. We also have a working example for that code. I think we should see that we're able to achieve that.